java/manager-build.xml

```diff
@@ -298,135 +302,156 @@
+    <exec executable="sh" outputproperty="podman.output">
+      <arg line="-c 'podman system connection ls -q | grep -w ${podman.connection.name}'"/>
     </exec>
 
-    <echo message="Linking the branding jar..." />
-    <exec failonerror="true" executable="kubectl">
-        <arg line="exec ${kubectl.args} -- mv ${deploy.dir}/WEB-INF/lib/java-branding.jar /usr/share/rhn/lib"/>
-    </exec>
+    <condition property="podman.connection.exists">
+      <length string="${podman.output}" when="greater" length="0"/>
+    </condition>
 
-    <exec failonerror="true" executable="kubectl">
-      <arg line="exec ${kubectl.args} -- ln -sf /usr/share/rhn/lib/java-branding.jar ${deploy.dir}/WEB-INF/lib/java-branding.jar"/>
-    </exec>
+    <echo message="Podman connection to '${podman.connection.name}' detected: ${podman.connection.exists}"/>
+  </target>
 
-    <echo message="Linking jars for Taskomatic..."/>
-    <exec failonerror="true" executable="kubectl">
-      <arg line="exec ${kubectl.args} -- ln -sf ${deploy.dir}/WEB-INF/lib/*.jar /usr/share/spacewalk/taskomatic"/>
+  <target name="add-podman-connection" depends="check-ssh-key,check-podman-connection" unless="podman.connection.exists" description="Adds podman system connection">
+    <echo message="Adding podman system connection..."/>
+    <exec executable="podman">
+      <arg line="system connection add --identity ${ssh.key} ${podman.connection.name} ssh://${deploy.user}@${podman.connection.name}.${deploy.host}"/>
     </exec>
-    <exec failonerror="true" executable="kubectl">
-      <arg line="exec ${kubectl.args} -- mv ${deploy.dir}/WEB-INF/lib/rhn.jar /usr/share/rhn/lib"/>
-    </exec>
-    <exec failonerror="true" executable="kubectl">
-      <arg line="exec ${kubectl.args} -- ln -sf /usr/share/rhn/lib/rhn.jar ${deploy.dir}/WEB-INF/lib"/>
+  </target>
+
+  <target name="remove-podman-connection" depends="warn-if-podman-not-installed" description="Removes podman system connection">
+    <echo message="Removing podman system connection..."/>
+    <exec executable="podman">
+      <arg line="system connection rm ${podman.connection.name}"/>
     </exec>
   </target>
 
-  <target name="deploy-static-resources-kube" depends="build-js,warn-if-kubectl-not-installed" description="Deploy css/js/img/font files in a running kubernetes pod">
-    <echo message="Copying frontend files to remote host..."/>
+  <target name="deploy-container" depends="webapp, add-podman-connection" description="Deploy a new copy of SUSE Manager in a running pod container">
+    <echo message="Copying files to pod container... ${build.dir} -> ${deploy.dir}" />
+
     <exec failonerror="true" executable="sh">
-        <arg line="-c 'tar c -p -C ${frontend.dist.dir} -f - . | kubectl exec -i ${kubectl.args} -- tar xf - -C /srv/www/htdocs'"/>
+      <arg line="-c 'tar c -C ${build.dir}/webapp -f - --exclude=log4j2.xml . | uyunictl exec -i -- tar xf - -C ${deploy.dir}'"/>
```

Comment:
I don't remember why I had to use `tar` here, but it seems that `kubectl` can copy folders too. In any case this should be handled at the `uyunictl` level, not the ant file to make `uyunictl` user's life easier. Is there any reason not to use your fancy `copy-to-container` macro?

Reply:
As we agreed, on this specific case we're sticking to the tar solution. The reason is that kubectl doesn't support excluding specific files/folder during the copy operation.